theano/gof/params_type.py

```diff
@@ -309,12 +518,22 @@ def c_support_code(self):
         sub = {'fail': '{this->setErrorOccurred(); return;}'}
         struct_name = self.name
         struct_name_defined = struct_name.upper()
+        c_support_code_set = set()
+        c_support_code_list = []
         c_declare_list = []
         c_init_list = []
         c_cleanup_list = []
         c_extract_list = []
         for attribute_name, type_instance in zip(self.fields, self.types):
 
+            try:
+                c_support_code_current = type_instance.c_support_code()
+                if c_support_code_current not in c_support_code_set:
+                    c_support_code_list.append(c_support_code_current)
+                    c_support_code_set.add(c_support_code_current)
```

Comment:
I find an ordered dict, but not an ordered set, into python collections.

I will use a set, and I will convert it to a sorted list just when necessary.